### PR TITLE
[MAINTENANCE] Add integration test for `MicrosoftTeamsNotificationAction`

### PR DESCRIPTION
**Pull request by cdkini**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by joshua-stauffer — CHANGES_REQUESTED** [review]
two suggestions, but happy to discuss further!
- can we add a test for at least one failure case? maybe revoke privileges from a webhook? 
- i don't think we gain a whole lot by putting every `Action` integration test into a single test, and it makes the test less legible. Would you consider renaming this to an individual test, with classes organized by action type?

File: tests/actions/conftest.py [hunk 1]
```diff
@@ -101,3 +104,10 @@ def sns(aws_credentials):
     with mock_sns():
         conn = boto3.client("sns")
         yield conn
+
+
+@pytest.fixture
+def mock_context(mocker: MockerFixture):
+    context = mocker.MagicMock(spec=AbstractDataContext)
+    set_context(context)
+    return context
```
Comment (cdkini) [inline tests/actions/conftest.py:113]: Moved to shared file

File: tests/actions/test_action_serialization.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    SlackNotificationAction,
+    SNSNotificationAction,
+    UpdateDataDocsAction,
+    ValidationAction,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TestActionSerialization:
```
Comment (cdkini) [inline tests/actions/test_action_serialization.py:23]: Serialization tests are moved to their own file just for better organization

File: tests/actions/test_core_actions.py [hunk 3]
```diff
@@ -53,198 +53,113 @@
...
-    def __init__(self, status_code: int, raise_for_status: bool = False):
-        self.status_code = status_code
-        self._raise_for_status = raise_for_status
-        self.text = "test_text"
-
-    def raise_for_status(self):
-        if self._raise_for_status:
-            raise requests.exceptions.HTTPError("test")
-
+# Global constants to be referenced in both tests and fixtures
```
Comment (cdkini) [inline tests/actions/test_core_actions.py:56]: Reorganized this whole file to have test classes based on action - I'll annotate where to look

File: tests/actions/test_core_actions.py [hunk 4]
```diff
@@ -593,6 +422,24 @@ def test_MicrosoftTeamsNotificationAction_run_webhook_substitution(
...
+        gx.get_context(mode="ephemeral")
+
+        action = MicrosoftTeamsNotificationAction(
+            name="test-action", teams_webhook="${GX_MS_TEAMS_WEBHOOK}"
+        )
+        result = action.run(checkpoint_result=checkpoint_result)
+        assert result == {
+            "microsoft_teams_notification_result": "Microsoft Teams notification succeeded."
+        }
+
```
Comment (cdkini) [inline tests/actions/test_core_actions.py]: This test is new

File: tests/actions/test_core_actions.py [hunk 5]
```diff
@@ -593,6 +422,46 @@ def test_MicrosoftTeamsNotificationAction_run_webhook_substitution(
...
+            name="test-action",
+            teams_webhook="${OUTDATED_MS_TEAMS_WEBHOOK}",  # Set as a secret in GH Actions
+        )
+        with caplog.at_level(logging.WARNING):
+            result = action.run(checkpoint_result=checkpoint_result)
+
+        assert result == {"microsoft_teams_notification_result": None}
+        assert caplog.records[-1].message.startswith(
+            "Request to Microsoft Teams API returned error"
+        )
```
Comment (cdkini) [inline tests/actions/test_core_actions.py]: These are the new tests

**Review by joshua-stauffer — APPROVED** [review]
thanks for addressing changes, looks great!

File: tests/actions/test_core_actions.py [hunk 6]
```diff
@@ -593,6 +422,47 @@ def test_MicrosoftTeamsNotificationAction_run_webhook_substitution(
...
+            teams_webhook="${OUTDATED_MS_TEAMS_WEBHOOK}",  # Set as a secret in GH Actions
+        )
+        with caplog.at_level(logging.WARNING):
+            result = action.run(checkpoint_result=checkpoint_result)
+
+        assert result == {"microsoft_teams_notification_result": None}
+        # Webhook was deleted, so we expect a 410 error
+        assert caplog.records[-1].message.startswith(
+            "Request to Microsoft Teams API returned error 410: 410 Client Error: Gone for url"
+        )
```
Comment (cdkini) [inline tests/actions/test_core_actions.py]: @billdirks

**Pull request merged**